provisioning/service/src/Contract/ContractApiHttp.cs

```diff
@@ -185,15 +185,26 @@ internal class ContractApiHttp : IContractApiHttp
 
         private static void ValidateHttpResponse(ContractApiResponse response)
         {
-            if (response.StatusCode >= HttpStatusCode.InternalServerError ||
-                (int)response.StatusCode == 429)
+            if (response.Body == null)
             {
-                throw new ProvisioningServiceClientHttpException(response, isTransient: true);
+                throw new DeviceProvisioningServiceException(response.ErrorMessage, response.StatusCode, response.Fields);
             }
-            else if (response.StatusCode >= HttpStatusCode.Ambiguous)
+
+            try
             {
-                throw new ProvisioningServiceClientHttpException(response, isTransient: false);
+                ResponseBody responseBody = JsonConvert.DeserializeObject<ResponseBody>(response.Body);
+
+                if (response.StatusCode >= HttpStatusCode.Ambiguous)
+                {
+                    throw new DeviceProvisioningServiceException(
+                        $"{response.ErrorMessage}:{response.Body}",
```

Comment:
```suggestion
                        $"{response.ErrorMessage}: {response.Body}",
```

Reply:
Actually, the error message might end in a period. I don't think a colon separating would look right.

Reply:
I tested it and noticed the `ErrorMessage` was short, not ending in a period, like "Not Found" if the registration Id was invalid while calling `GetIndividualEnrollmentAsync(string registrationId, CancellationToken cancellationToken)`. 

With that being said, I can stop using colon here just in cases that some other error message ends in a period.

Reply:
Then it is probably fine. I don't care **that** much.